### DISCUSSION TITLE
MeshCentral Agent Customization

### DISCUSSION THREAD

**Discussion by primetecabt** [Ylianst/MeshCentral#6356] [Q&A] [answered]
First of all, I would like to thank you for the excellent project. I have been implementing MeshCentral for about 3 months, and with each step I learn a new feature. I recently came across the agent customization, but I can't get it to work properly. Sometimes I get the message "Server agentcustomization has no users, next new account will be site administrator", but I don't know if it is an error or a warning.

Here is my config.json in case anyone can help me.

```

{
  "$schema": "http://info.meshcentral.com/downloads/meshcentral-config-schema.json",
  "settings": {
    "cert": "remoto.dominio.com",
    "_WANonly": true,
    "_LANonly": true,
    "sessionKey": "G13dn4ss1Ad07MpYzrU3ncQsDm8Ubf79",
    "port": 443,
    "_aliasPort": 443,
    "redirPort": 80,
    "_redirAliasPort": 80,
    "AgentPong": 300,
    "TLSOffload": false,
    "SelfUpdate": false,
    "AllowFraming": "false",
    "WebRTC": "true",
    "AutoBackup": {
      "backupPath": "/opt/meshcentral/meshcentral-backups",
      "backupInvervalHours": 24,
      "keepLastDaysBackup": "",
      "zippassword": ""
    }
  },
  "domains": {
    "": {
      "title": "Prime Remoto",
      "loginPicture": "logo_inicial.png",
      "_title2": "Servername",
      "_minify": true,
      "NewAccounts": "false",
      "_userNameIsEmail": true,
      "certUrl": "https://remoto.dominio.com"
    },
      "agentCustomization": {
      "displayName": "Prime Tecnologia Remoto",
      "description": "Serviço em Segundo Plano do Agente MeshCentral",
      "companyName": "Prime Tecnologia",
      "serviceName": "Serviço do Agente MeshCentral",
      "image": "logo_inicial.png",
      "fileName": "meshagent",
      "foregroundColor": "#FFA500",
      "backgroundColor": "#EE82EE"
    }
  },
 
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before>",
    "_email": "myemail@mydomain.com",
    "_names": "myserver.mydomain.com",
    "production": false
  }
}

```

**The domain remoto.dominio.com is merely illustrative**

**Comment by si458** [comment, marked as answer]
You haven't formatted ur json correctly.
Always use https://jsonlint.com and put ur json into it and let it format it so its viewable correctly 

You are suppose to put the `agentCustomization` section inside of your domain section.

So in the default case it's inside the `""` section

**Comment by primetecabt** [comment]
Thank you very much my friend, with your tips I was able to customize the agent. Now I'm working on customizing the icon of the generated executable.
